internal/proxy/task_index.go

```diff
@@ -0,0 +1,678 @@
+
+	"github.com/milvus-io/milvus/internal/common"
+	"github.com/milvus-io/milvus/internal/log"
+	"github.com/milvus-io/milvus/internal/types"
+
+	"github.com/milvus-io/milvus/api/commonpb"
+	"github.com/milvus-io/milvus/api/milvuspb"
+	"github.com/milvus-io/milvus/api/schemapb"
+	"github.com/milvus-io/milvus/internal/util/funcutil"
+	"github.com/milvus-io/milvus/internal/util/indexparamcheck"
+	"github.com/milvus-io/milvus/internal/util/typeutil"
+)
+
+const (
+	CreateIndexTaskName           = "CreateIndexTask"
+	DescribeIndexTaskName         = "DescribeIndexTask"
+	DropIndexTaskName             = "DropIndexTask"
+	GetIndexStateTaskName         = "GetIndexStateTask"
+	GetIndexBuildProgressTaskName = "GetIndexBuildProgressTask"
+
+	AutoIndexName = "AUTOINDEX"
+	DimKey        = common.DimKey
+)
+
+type createIndexTask struct {
+	Condition
+	req        *milvuspb.CreateIndexRequest
+	ctx        context.Context
+	rootCoord  types.RootCoord
+	indexCoord types.IndexCoord
+	result     *commonpb.Status
+
+	isAutoIndex     bool
+	userIndexParams []*commonpb.KeyValuePair
+
+	collectionID UniqueID
+	fieldSchema  *schemapb.FieldSchema
+}
+
+func (cit *createIndexTask) TraceCtx() context.Context {
+	return cit.ctx
+}
+
+func (cit *createIndexTask) ID() UniqueID {
+	return cit.req.GetBase().GetMsgID()
+}
+
+func (cit *createIndexTask) SetID(uid UniqueID) {
+	cit.req.GetBase().MsgID = uid
+}
+
+func (cit *createIndexTask) Name() string {
+	return CreateIndexTaskName
+}
+
+func (cit *createIndexTask) Type() commonpb.MsgType {
+	return cit.req.GetBase().GetMsgType()
+}
+
+func (cit *createIndexTask) BeginTs() Timestamp {
+	return cit.req.GetBase().GetTimestamp()
+}
+
+func (cit *createIndexTask) EndTs() Timestamp {
+	return cit.req.GetBase().GetTimestamp()
+}
+
+func (cit *createIndexTask) SetTs(ts Timestamp) {
+	cit.req.Base.Timestamp = ts
+}
+
+func (cit *createIndexTask) OnEnqueue() error {
+	cit.req.Base = &commonpb.MsgBase{}
+	return nil
+}
+
+func (cit *createIndexTask) parseIndexParams() error {
+	indexParams := cit.req.GetExtraParams()
+	cit.userIndexParams = append(cit.userIndexParams, indexParams...)
+	isVecIndex := typeutil.IsVectorType(cit.fieldSchema.DataType)
+	if !isVecIndex {
+		if cit.fieldSchema.DataType == schemapb.DataType_VarChar {
+			indexParams = append(indexParams, &commonpb.KeyValuePair{Key: common.IndexTypeKey, Value: DefaultStringIndexType})
+		} else {
+			indexParams = append(indexParams, &commonpb.KeyValuePair{Key: common.IndexTypeKey, Value: DefaultIndexType})
+		}
+	}
+
+	indexParamsMap := make(map[string]string)
+	for _, kv := range indexParams {
+		if kv.Key == common.IndexParamsKey {
+			params, err := funcutil.ParseIndexParamsMap(kv.Value)
+			if err != nil {
+				return err
+			}
+			for k, v := range params {
+				indexParamsMap[k] = v
+			}
+		} else {
+			indexParamsMap[kv.Key] = kv.Value
+		}
+	}
+
+	if isVecIndex {
+		specifyIndexType, exist := indexParamsMap[common.IndexTypeKey]
+		if Params.AutoIndexConfig.Enable {
```

Comment:
if Params.AutoIndexConfig.Enable not true, where did we check index type and fill in default index?